maplibre-adapter/src/main/java/com/amplifyframework/geo/maplibre/AmplifyMapLibreAdapter.kt

```diff
@@ -16,70 +16,45 @@
 package com.amplifyframework.geo.maplibre
 
 import android.content.Context
-import androidx.annotation.VisibleForTesting
-
-import com.amazonaws.auth.AWSCredentialsProvider
-import com.amplifyframework.auth.AuthCategory
 import com.amplifyframework.core.Amplify
 import com.amplifyframework.geo.GeoCategory
-import com.amplifyframework.geo.maplibre.http.AWS4SigningInterceptor
+import com.amplifyframework.geo.location.AWSLocationGeoPlugin
+import com.amplifyframework.geo.maplibre.http.AWSRequestSignerInterceptor
 import com.amplifyframework.geo.models.MapStyle
 import com.amplifyframework.geo.options.GetMapStyleDescriptorOptions
-
 import com.mapbox.mapboxsdk.Mapbox
 import com.mapbox.mapboxsdk.WellKnownTileServer
 import com.mapbox.mapboxsdk.maps.MapboxMap
 import com.mapbox.mapboxsdk.maps.Style
 import com.mapbox.mapboxsdk.module.http.HttpRequestUtil
-import kotlinx.coroutines.*
+import kotlinx.coroutines.MainScope
+import kotlinx.coroutines.launch
 import okhttp3.OkHttpClient
 
-/**
- * Entry point to initialize MapLibre instance and configure it to be
- * authorized to make requests directly to Amazon Location Service.
- */
-object AmplifyMapLibreAdapter {
-    private const val COGNITO_AUTH_PLUGIN_KEY = "awsCognitoAuthPlugin"
-    private const val GEO_SERVICE_NAME = "geo"
+private const val LOCATION_GEO_PLUGIN_KEY = "awsLocationGeoPlugin"
 
-    private val log = Amplify.Logging.forNamespace("amplify:maplibre-adapter")
+class AmplifyMapLibreAdapter(
```

Comment:
I understand the benefits of having the adapter be instantiable, but my only concern is that `Mapbox` itself is a singleton. Individually instantiable adapter might erroneously lead developers to think that they can have multiple differently configured adapters when in fact each one will be using the exact same copy of Mapbox instance.

Reply:
Developers never instantiate the adapter directly, the MapView component itself does. I think the open question here is if the adapter should even be exposed directly.

This loose dependency between an adapter that needs to be setup before the component is initialized is error-prone and unnecessary. My proposed solution fixes that.

What we can do is to make the constructor internal and expose a static singleton method.